src/Symfony/Component/Messenger/Worker.php

```diff
@@ -35,22 +35,22 @@ public function __construct(ReceiverInterface $receiver, MessageBusInterface $bu
      */
     public function run()
     {
-        $iterator = $this->receiver->receive();
+        if (function_exists('pcntl_signal')) {
+            pcntl_signal(SIGTERM, function () {
+                $this->receiver->stop();
+            });
+        }
+
+        $this->receiver->receive(function($message) {
+            if (null === $message) {
+                return;
+            }
```

Comment:
Because the receiver's `receive` callback can receive a `null` message in case of timeout. This is to tackle [the point exposed by @lyrixx](https://github.com/symfony/symfony/pull/26632#discussion_r178811238): that way, you can add extra capabilities (like sending pings to the monitoring, etc...) at timely intervals even if no messages are handled. You basically put a `read_timeout` on the connection and decorate the receiver to do whatever you want (with or without messages coming in).